### PR TITLE
Fix calculating stack size under musl for the main thread

### PR DESCRIPTION
**Pull request by azat**
This should fix the stack size for the main thread.

### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
ClickHouse is built with Musl instead of GLibc

On top of: #52550 (cc @alexey-milovidov )

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit 97baa7cbfa5dbc58f3e1fe071afd52e575dd3b08 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52721/97baa7cbfa5dbc58f3e1fe071afd52e575dd3b08/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by alexey-milovidov — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -374,6 +374,11 @@ option(ENABLE_LIBRARIES "Enable all external libraries by default" ON)
 # Increase stack size on Musl. We need big stack for our recursive-descend parser.
 if (USE_MUSL)
     set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,stack-size=2097152")
+
+    # musl uses tricky code to detect the stack size of the main thread with
+    # mremap(), but with stack protector you will get EFAULT.
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-stack-protector")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-stack-protector")
```
Comment (azat) [inline CMakeLists.txt:381]: This patch should be reverted.

**Comment by azat** [comment]
The last patch fixes the problem, but now the problem is that unwind is slow and `01304_polygons_sym_difference`.

**Comment by azat** [comment]
FWIW I've tried to the following things:
- update musl
- compile musl with debug
- compile musl with debug and `-fasynchronous-unwind-tables`

None of this changes anything so far.

**Comment by alexey-milovidov** [comment]
#52769

**Comment by alexey-milovidov** [comment]
What particular place of unwinding is slow?

**Comment by azat** [comment]
>What particular place of unwinding is slow?

```
Overhead  Command          Shared Object     Symbol
  46.10%  HTTPHandler      clickhouse        [.] libunwind::CFI_Parser<libunwind::LocalAddressSpace>::parseCIE
  29.69%  HTTPHandler      clickhouse        [.] libunwind::LocalAddressSpace::getEncodedP
  17.57%  HTTPHandler      clickhouse        [.] libunwind::CFI_Parser<libunwind::LocalAddressSpace>::findFDE
   6.00%  HTTPHandler      clickhouse        [.] libunwind::LocalAddressSpace::getULEB128
   0.03%  HTTPHandler      clickhouse        [.] libunwind::CFI_Parser<libunwind::LocalAddressSpace>::parseFDEInstructions
   0.02%  HTTPHandler      clickhouse        [.] libunwind::DwarfInstructions<libunwind::LocalAddressSpace, libunwind::Registers_x86_64>::stepWithDwarf
```

**Comment by azat** [comment]
So this PR now contains only fixes and build env for musl (since it cannot be enabled due to performance reason yet).

**Review by alexey-milovidov — APPROVED** [review]

**Comment by azat** [comment]
> test_host_regexp_multiple_ptr_records_concurrent/test.py::test_host_regexp_multiple_ptr_v4

Known issie

**Pull request merged**